src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/SchemaFactory.cs

```diff
@@ -132,5 +122,22 @@ public void RegisterValueConverter(IAstFromValueConverter converter)
         {
             _schema.Value.RegisterValueConverter(converter);
         }
+
+        public TType GetMetadata<TType>(string key, TType defaultValue = default(TType))
+        {
+            throw new NotImplementedException();
+        }
+
+        public TType GetMetadata<TType>(string key, Func<TType> defaultValueFactory)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool HasMetadata(string key)
+        {
+            throw new NotImplementedException();
+        }
+
+        public IDictionary<string, object> Metadata { get; }
```

Comment:
Probably here you could implement and initialize this property to avoid possible exceptions if some code (maybe even inner GraphQL) would call these interface methods.
```
public IDictionary<string, object> Metadata { get; private set;} = new Dictionary<string, object>();
```
and using this dictionary implement 3 NotImplemented methods, like this
```
        public TType GetMetadata<TType>(string key, TType defaultValue = default(TType))
        {
            return Metadata.ContainsKey(key) ? Metadata[key] : defaultValue;
        }
```

Reply:
updated

Reply:
done